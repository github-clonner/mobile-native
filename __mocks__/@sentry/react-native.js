export const captureMessage = jest.fn();
export default {};
