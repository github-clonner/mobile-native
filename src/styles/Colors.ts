//@ts-nocheck
export default colors = {
  primary: '#4690D6',
  secondary: '#FFDD63',
  tertiary: '#4C9EB1',
  danger: '#f53d3d',
  light: '#f7f7f7',
  medium: '#b0bec5',
  dark: '#222222',
  greyed: '#cccccc',
  lightGreyed: '#d8d8d8',
  darkGreyed: '#777777',
  explicit: '#f44336',
  action: 'rgb(96, 125, 139)',
};

export const LIGHT_THEME = {
  name: 'LIGHT',
  primary_background: '#F5F5F5',
  secondary_background: '#FFFFFF',
  tertiary_background: '#E3E4E9',
  primary_text: '#4F4F50',
  secondary_text: '#7D7D82',
  tertiary_text: '#9B9B9B',
  primary_button: '#5DBAC0',
  primary_border: '#DCE2E4',
  separator: '#323C3F',
  done: '#4C92A4',
  action: '#A5A5A5',
  icon: '#7D7D82',
  icon_active: '#0085DD',
  link: '#0091FF',
  alert: '#E03C20',
  green: 'rgb(76, 175, 80)',
};

export const DARK_THEME = {
  name: 'DARK',
  primary_background: '#252E31',
  secondary_background: '#202527',
  tertiary_background: '#404E53',
  primary_text: '#FFFFFF',
  secondary_text: '#AEB0B8',
  tertiary_text: '#797B82',
  primary_button: '#5DBAC0',
  primary_border: '#404A4E',
  separator: '#323C3F',
  done: '#4C92A4',
  action: '#A5A5A5',
  icon: '#AEB0B8',
  icon_active: '#0085DD',
  link: '#0091FF',
  alert: '#E03C20',
  green: 'rgb(76, 175, 80)',
};
