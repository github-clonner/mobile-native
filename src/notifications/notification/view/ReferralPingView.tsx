//@ts-nocheck
import React, { PureComponent } from 'react';

import { Text, View } from 'react-native';

import i18n from '../../../common/services/i18n.service';

/**
 * Referral Ping
 */
export default class ReferralPingView extends PureComponent {
  /**
   * Navigate to navToContributions
   */
  navToContributions = () => {
    this.props.navigation.push('Contributions');
  };

  /**
   * Render
   */
  render() {
    const styles = this.props.styles;

    return (
      <View style={styles.bodyContents}>
        <Text onPress={this.navToContributions}>
          {i18n.to('notification.referralPing', null, {
            user: (
              <Text style={styles.link}>{this.props.entity.fromObj.name}</Text>
            ),
          })}
        </Text>
      </View>
    );
  }
}
