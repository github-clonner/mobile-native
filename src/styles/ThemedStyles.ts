//@ts-nocheck
import { StyleSheet, Platform } from 'react-native';
import { observable, action, reaction } from 'mobx';

import { DARK_THEME, LIGHT_THEME } from './Colors';
import { DefaultTheme, DarkTheme } from '@react-navigation/native';
import { buildStyle } from './Style';

import type { ThemedStyle } from './Style';

/**
 * ThemedStylesStore
 */
class ThemedStylesStore {
  /**
   * Theme observable
   * 1 Dark
   * 0 Light
   * -1 Not loaded
   * @property {Observable<numeric>}
   */
  @observable theme: number = -1;

  navTheme?: object = undefined;
  defaultScreenOptions?: object = undefined;

  /**
   * Style
   */
  style: ThemedStyle = {} as ThemedStyle;

  /**
   * Initialice themed styles
   */
  async init() {
    // load stored theme value here
    this.generateStyle();
  }

  /**
   * Set dark theme
   */
  @action
  setDark() {
    this.theme = 1;
    this.generateStyle();
  }

  /**
   * Set light theme
   */
  @action
  setLight() {
    this.theme = 0;
    this.generateStyle();
  }

  /**
   * Set theme
   * @param {number} value
   */
  @action
  setTheme(value) {
    this.theme = value;
    this.generateStyle();
  }

  /**
   * On theme change reaction
   * @param {Function} fn
   */
  onThemeChange(fn) {
    return reaction(
      () => [this.theme],
      async (args) => await fn(...args),
      {
        fireImmediately: false,
      },
    );
  }

  /**
   * Get color of theme based on property
   * @param {String} prop
   */
  getColor(prop) {
    const theme = this.theme ? DARK_THEME : LIGHT_THEME;
    return theme[prop];
  }

  /**
   * Generates the current theme
   */
  generateStyle() {
    const theme = this.theme ? DARK_THEME : LIGHT_THEME;

    const baseTheme = this.theme === 0 ? DefaultTheme : DarkTheme;

    this.navTheme = {
      ...baseTheme,
      colors: {
        ...baseTheme.colors,
        background: 'transparent',
        // card: theme.backgroundSecondary, // generates an error on ios
        text: theme.primary_text,
        primary: theme.icon,
      },
    };

    this.defaultScreenOptions = {
      title: '',
      headerStyle: {
        backgroundColor: theme.secondary_background,
      },
      contentStyle: {
        backgroundColor: theme.primary_background,
      },
      stackAnimation: Platform.select({
        ios: 'default',
        android: 'fade',
      }),
    };

    this.style = StyleSheet.create(buildStyle(theme));
  }
}

export default new ThemedStylesStore();
