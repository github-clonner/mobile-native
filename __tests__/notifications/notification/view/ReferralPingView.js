import 'react-native';
import React from 'react';
import { Text, TouchableOpacity } from "react-native";
import { shallow } from 'enzyme';
import ReferralPingView from '../../../../src/notifications/notification/view/ReferralPingView';
import styles from '../../../../src/notifications/notification/style';

// fake data generation
import boostNotificationFactory from '../../../../__mocks__/fake/notifications/BoostNotificationFactory';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {

  const entity = boostNotificationFactory('referral_ping');

  const notification = renderer.create(
    <ReferralPingView styles={styles} entity={entity}/>
  ).toJSON();

  expect(notification).toMatchSnapshot();
});