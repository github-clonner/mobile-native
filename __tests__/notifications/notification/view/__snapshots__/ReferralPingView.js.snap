// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "flexWrap": "wrap",
    }
  }
>
  <Text
    onPress={[Function]}
  >
    Free tokens are waiting for you! Once you join the rewards program by setting up your Minds wallet, both you and 
    <Text
      style={
        Object {
          "fontWeight": "600",
        }
      }
    >
      someUser
    </Text>
     will earn tokens for your referral
  </Text>
</View>
`;
